### PR TITLE
Implement File Table

### PR DESCRIPTION
**Pull request by leon-fw**
Added module, controller, entity and service for the **File** table.

Notes:
- Apparently there is no BLOB type in Postgres, but BYTEA should be close enough.
- ER Diagram: https://miro.com/app/board/uXjVPbOq9NI=/

### CODE REVIEW

File: server/src/application/application.entity.ts [hunk 1]
```diff
@@ -47,4 +48,7 @@ export class Application extends CustomBaseEntity {
 
   @ManyToOne(() => User, (user) => user.applications)
   user: User;
+
+  @OneToMany(() => File, (file) => file.application)
+  files: File[];
```
Comment (arranfw) [inline server/src/application/application.entity.ts:53]: I'm wondering if `File` is a specific enough name? Would something like `SupportingDocument` be better? What do you think?

File: server/src/file/file.entity.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+@Entity()
+export class File extends CustomBaseEntity {
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
+
+  @Column({ type: 'varchar', length: 100, nullable: false })
+  name: string;
+
+  @Column({ type: 'bytea', nullable: false })
```
Comment (arranfw) [inline server/src/file/file.entity.ts:13]: Just double checking we're intending to store files directly in the database rather than in something like S3?

**Review by arranfw — APPROVED** [review]

File: server/src/file/file.entity.ts (hunk 2 above)
Comment (leon-fw) [reply server/src/file/file.entity.ts:13]: Karthikk said to implement it like this for now, so I guess it's to speed thing us/for convenience for now. If there's a better way - let me know!

File: server/src/application/application.entity.ts (hunk 1 above)
Comment (karthikktamilmani-freshworks) [reply server/src/application/application.entity.ts:53]: yeah can be renamed. Makes sense!

File: server/src/file/file.entity.ts (hunk 2 above)
Comment (karthikktamilmani-freshworks) [reply server/src/file/file.entity.ts:13]: I'll confirm in some time!

**Pull request merged**